appium/webdriver/appium_service.py

```diff
@@ -215,38 +119,44 @@ def start(self, **kwargs: Any) -> sp.Popen:
             error_msg = startup_failure_msg
         if error_msg is not None:
             if stderr == sp.PIPE and self._process.stderr is not None:
+                # noinspection PyUnresolvedReferences
                 err_output = self._process.stderr.read()
                 if err_output:
                     error_msg += f'\nOriginal error: {str(err_output)}'
             self.stop()
             raise AppiumServiceError(error_msg)
         return self._process
 
-    def stop(self) -> bool:
+    def stop(self, timeout: float = 5.5) -> bool:
         """Stops Appium service if it is running.
 
         The call will be ignored if the service is not running
         or has been already stopped.
 
-        Returns:
-            `True` if the service was running before being stopped
+        :param timeout: The maximum time in float seconds to wait
+        for the server process to terminate
+        :return: `True` if the service was running before being stopped
         """
-        is_terminated = False
+        was_running = False
         if self.is_running:
             assert self._process
+            was_running = True
             self._process.terminate()
-            self._process.communicate(timeout=5)
-            is_terminated = True
+            try:
+                self._process.communicate(timeout=timeout)
+            except sp.SubprocessError:
+                import signal
+
+                os.kill(self._process.pid, signal.SIGKILL)
```

Comment:
Would `self._process.kill()` be better for cross platform?
https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate

Reply:
process.kill is just an alias to process.terminate, which is already called above

Reply:
for Win? i though for Linux are macOS are sigkill in `kill` and alias to terminate for Win. So in terms of Win, yea, it's just an alias already called, so it calls  terminal twice thing

Reply:
Is it in `if _mswindows:`?

![Screenshot 2024-10-03 at 10 43 08 AM](https://github.com/user-attachments/assets/19bae242-105b-46ef-beb8-c2ab50d2bb1a)

Reply:
I did not pay attention to the `else:` block. Changed to use .kill